client-src/elements/autolink.ts

```diff
@@ -118,9 +124,9 @@ function replaceLinkRef(match) {
   }
   LINK_TRAILING_CHARS.forEach(([begin, end]) => {
     if (content.endsWith(end)) {
```

Comment:
When you're adding `!` annotations, consider whether you can get the type checker to figure it out without the annotation. In this case, it's easy enough to

```suggestion
    if (end && content.endsWith(end)) {
```